### DISCUSSION TITLE
Support for APDS9930

### DISCUSSION THREAD

**Discussion by Basti149** [arendst/Tasmota#22154] [Ideas]
The APDS9930 sensor does not work with the APDS9960 driver so I wanted to ask if someone can integrate this sensor into Tasmota. I am happy to publish all relevant data and also help with development.

But still nothing is displayed in the console
This is only displayed here with i2cscan.

15:46:20.893 RSL: RESULT = {"I2CScan":"Device(s) found at 0x39"}
15:46:42.869 CMD: Sensor 27 1
15:46:42.873 RSL: RESULT = {"Command":"Error"}

Do I have to enter anything else? I use Tasmota 14.x.x

![grafik](https://github.com/user-attachments/assets/aff4c24b-b96d-4338-a4f0-7f1d98113693)

**Comment by sfromis** [comment]
Don't just open another thread, and "Issues" is not for asking questions. I did [respond](https://github.com/arendst/Tasmota/issues/7346#issuecomment-2353289783) in the other thread, and what you post now is not a valid command, should be `Sensor27 1`, but only if the sensor is actually included in the binary you're running, and detecting the sensor.

**Comment by Basti149** [comment]
This is a different sensor than the APD9960, I meant the APD9930 and this is not yet supported by Tasmota. That's why I want someone to integrate the sensor into Tasmota.

[Image]



That's why he couldn't detect the sensor

https://components101.com/sensors/apds-9930-digital-ambient-light-and-proximity-sensor

https://github.com/depau/APDS9930

**Comment by Basti149** [comment]
What should I do so that it will be taken into account in the next Tasmota version? I would also provide all parameters.

Holen Sie sich Outlook für Android<https://aka.ms/ghei36>
________________________________

**Reply by barbudor** [reply]
What do you mean 'taken into account in next Tasmota version'?

If the driver for APDS9930 does not exist some with interrest has to code it and add it to Tasmota
Could it be you ?

**Comment by Basti149** [comment]
@barbudor 
Unfortunately my programming language is not that good yet. I can manage it but I would agree to test it and provide all the necessary data.
The question is whether the programming is identical to the APD9600
